crates/trie/sparse/src/parallel_trie.rs

```diff
@@ -71,10 +80,50 @@ impl ParallelSparseTrie {
         node: TrieNode,
         masks: TrieMasks,
     ) -> SparseTrieResult<()> {
-        let _path = path;
-        let _node = node;
-        let _masks = masks;
-        todo!()
+        // TODO parallelize
+        if let Some(subtrie) = self.lower_subtrie_for_path(&path) {
+            return subtrie.reveal_node(path, &node, masks);
+        }
+
+        // If there is no subtrie for the path it means the path is 2 or less nibbles, and so
+        // belongs to the upper trie.
+        self.upper_subtrie.reveal_node(path.clone(), &node, masks)?;
+
+        // The previous upper_trie.reveal_node call will not have revealed any child nodes via
+        // reveal_node_or_hash if the child node would be found on a lower subtrie. We handle that
+        // here by manually checking the specific cases where this could happen, and calling
+        // reveal_node_or_hash for each.
+        match node {
+            TrieNode::Branch(branch) => {
+                // If a branch is at the second level of the trie then it will be in the upper trie,
+                // but all of its children will be in the lower trie.
+                if path.len() == 2 {
```

Comment:
this feels like a magic number, let's put it in the constant? And add a comment that adjustments to `SparseSubtrieType::from_path` are needed if one wants to change the constant value.

Reply:
Wdyt if I do this change as a separate PR? There's a bunch of places we could remove the magic number usage, not all of them introduced here, would be good to keep the scope of this from getting larger imo